cli/tools/compile.rs

```diff
@@ -170,31 +170,34 @@ async fn resolve_compile_executable_output_path(
   let module_specifier =
     resolve_url_or_path(&compile_flags.source_file, current_dir)?;
 
-  let mut output = compile_flags.output.clone();
-
-  if let Some(out) = output.as_ref() {
+  let output_flag = compile_flags.output.clone();
+  let mut output_path = if let Some(out) = output_flag.as_ref() {
+    let mut out_path = PathBuf::from(out);
     if path_has_trailing_slash(out) {
```

Comment:
Maybe just:

```suggestion
    if out.ends_with('/') || out.ends_with('\\') {
```

Then we can delete `path_has_trailing_slash`. `path_has_trailing_slash` seems a little sketchy because on Windows people often just use a forward slash anyway.